internal/it/serialization.go

```diff
@@ -17,13 +17,59 @@
 package it
 
 import (
+	"context"
+	"fmt"
 	"testing"
 
 	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
 )
 
 func SerializationTester(t *testing.T, f func(t *testing.T, config hazelcast.Config, clusterID, mapName string)) {
 	ensureRemoteController(true)
 	mapName := NewUniqueObjectName("map")
 	f(t, defaultTestCluster.DefaultConfig(), defaultTestCluster.ClusterID, mapName)
 }
+
+func inMemoryFormatConfig(clusterName, mapName, inMemoryFormat string, port int32) string {
+	return fmt.Sprintf(`
+        <hazelcast xmlns="http://www.hazelcast.com/schema/config"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://www.hazelcast.com/schema/config
+            http://www.hazelcast.com/schema/config/hazelcast-config-4.0.xsd">
+            <cluster-name>%s</cluster-name>
+            <map name="%s">
+				<in-memory-format>%s</in-memory-format>
+			</map>
+			<network>
+               <port>%d</port>
+            </network>
+        </hazelcast>
+	`, clusterName, mapName, inMemoryFormat, port)
+}
+
+func CompactFormatTester(t *testing.T, mapName string, compactSerializers []serialization.CompactSerializer, port int, f func(t *testing.T, client1, client2 *hazelcast.Client)) {
+	ensureRemoteController(false)
+	testCases := []struct {
+		name           string
+		inMemoryFormat string
+	}{
+		{name: "inMemoryFormat:BINARY", inMemoryFormat: "BINARY"},
+		//todo: uncomment this after metadata distribution is implemented {name: "inMemoryFormat:OBJECT", inMemoryFormat: "OBJECT"},
+	}
+	for _, tc := range testCases {
+		testName := t.Name() + tc.name
+		t.Run(testName, func(t *testing.T) {
+			tc := StartNewClusterWithConfig(2, inMemoryFormatConfig(testName, mapName, tc.inMemoryFormat, int32(port)), port)
```

Comment:
🤔 Why? Let's also change the Java test then

Reply:
https://stackoverflow.com/a/64058131/3782459

Reply:
Let's use two members as a result of the standup discussion.  Not one because to make it similar to java. (schema dist. or partitions related stuff etc. should be the same )